apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,61 +1,55 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { NgFor, NgIf, NgTemplateOutlet } from '@angular/common';
+import {
+  Component,
+  ContentChild,
+  Directive,
+  EventEmitter,
+  Input,
+  Output,
+  TemplateRef,
+} from '@angular/core';
 import { ListItemComponent } from '../list-item/list-item.component';
 
+@Directive({
+  selector: 'app-card-image',
+  standalone: true,
+})
+export class CardImageDirective {}
+
+@Directive({
+  selector: '[appCardListItem]',
+  standalone: true,
+})
+export class CardListItemDirective<T> {
+  constructor(public templateRef: TemplateRef<{ $implicit: T }>) {}
+}
+
 @Component({
   selector: 'app-card',
   template: `
-    <div
-      class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4"
-      [class]="customClass">
-      <img
-        *ngIf="type === CardType.TEACHER"
-        src="assets/img/teacher.png"
-        width="200px" />
-      <img
-        *ngIf="type === CardType.STUDENT"
-        src="assets/img/student.webp"
-        width="200px" />
+    <div class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4">
+      <ng-content select="app-card-image"></ng-content>
 
       <section>
-        <app-list-item
-          *ngFor="let item of list"
-          [name]="item.firstName"
-          [id]="item.id"
-          [type]="type"></app-list-item>
+        @for (item of list; track $index) {
+          <ng-template
+            [ngTemplateOutlet]="rowTemplate.templateRef"
+            [ngTemplateOutletContext]="{ $implicit: item }"></ng-template>
+        }
       </section>
 
       <button
         class="rounded-sm border border-blue-500 bg-blue-300 p-2"
-        (click)="addNewItem()">
+        (click)="actionClicked.emit()">
         Add
       </button>
     </div>
   `,
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  imports: [NgIf, NgFor, ListItemComponent, NgTemplateOutlet],
 })
-export class CardComponent {
-  @Input() list: any[] | null = null;
-  @Input() type!: CardType;
-  @Input() customClass = '';
-
-  CardType = CardType;
-
-  constructor(
-    private teacherStore: TeacherStore,
-    private studentStore: StudentStore,
-  ) {}
-
-  addNewItem() {
-    if (this.type === CardType.TEACHER) {
-      this.teacherStore.addOne(randTeacher());
-    } else if (this.type === CardType.STUDENT) {
-      this.studentStore.addOne(randStudent());
-    }
-  }
+export class CardComponent<T> {
+  @Input() list: T[] | null = null;
+  @Output() actionClicked = new EventEmitter();
+  @ContentChild(CardListItemDirective) rowTemplate!: CardListItemDirective<T>;
```

Comment:
awesome work

Reply:
Thank you for taking the time to review it